File: AndroidEdu/AppsForDemo/BasicGraphics/app/src/main/java/edu/ucsb/ece150/demos/basicgraphics/MainActivity.java
package edu.ucsb.ece150.demos.basicgraphics;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        final GraphicsImageView myImageView = (GraphicsImageView) findViewById(R.id.my_image_view);
        final Button drawButton = (Button) findViewById(R.id.button_draw);
        drawButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                myImageView.draw();
            }
        });

        final Button clearButton = (Button) findViewById(R.id.button_clear);
        clearButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                myImageView.clear();
            }
        });
    }
}

File: AndroidEdu/KotlinPlygrd/app/src/main/kotlin/edu/ucsb/demos/xuanwang/kotlinplayground/MainActivity.kt
package edu.ucsb.demos.xuanwang.kotlinplayground

import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import org.jetbrains.anko.*
import org.jetbrains.anko.sdk25.coroutines.onClick

class MainActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)

    verticalLayout {
      val mText = textView {
        text = "Welcome to Kotlin"
        textSize = 20f
      }

      val mInput = editText {
        hint = "Please enter some text"
      }

      button("change text") {
        onClick {
          mText.text = mInput.text
        }
      }.lparams(width = matchParent) {
        horizontalMargin = dip(5)
        topMargin = dip(10)
      }
    }
  }
}

